text/0000-command-ergonomics.md

```diff
@@ -0,0 +1,542 @@
+- Feature Name: (fill me in with a unique ident, `my_awesome_feature`)
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Improve the ergonomics of `std::process::Command`,
+and better protect programmers from error handling mistakes.
+
+This RFC is principally an attempt to
+gain consensus on a fleshed out version of
+the MR [#89004](https://github.com/rust-lang/rust/pull/89004)
+which proposed to add some convenience APIs to `std::process::Command`.
+
+# Motivation
+[motivation]: #motivation
+
+The current API for `std::process::Command`
+makes it unnecessarily difficult to perform common tasks,
+such as "simply running a program",
+and running a program and collecting its output.
+
+The APIs that are currently provided invite mistakes
+(for example, failing check the exit status in `Output`,
+and deadlock errors).
+Writing code that is fully correct
+and produces good error messages is cumbersome,
+and sometimes subtle.
+
+Running subprocesses is inherently complex, and provides many opportunites for errors to occur - so there are some questions about how to best represent this complexity in a convenient API.
+
+Existing work in this area has been fragmented into a number of 
+MRs and issues, making it hard to see the wood for the trees,
+and has become bogged down due to lack of decision
+on the overall approach.  Some references:
+
+ * [#84908](https://github.com/rust-lang/rust/issues/84908)
+   Stabilisation tracking issue for `ExitStatusError` 
+ * [#81452](https://github.com/rust-lang/rust/pull/81452)
+   MR, closed as blocked: Add `#[must_use]` to `process::Command`, `process::Child` and `process::ExitStatus` 
+ * [#89004](https://github.com/rust-lang/rust/pull/89004)
+   MR, closed: Add convenience API to `std::process::Command`
+ * [#88306](https://github.com/rust-lang/rust/pull/88306)
+   MR, closed: `ErrorKind::ProcessFailed` and `impl From<ExitStatusError>`
+ * [#93565](https://github.com/rust-lang/rust/pull/93565)
+   MR, draft: `impl Try for ExitStatus`
+ * [#73126](https://github.com/rust-lang/rust/issues/73126)
+   issue: `std::process::Command` `output()` method error handling hazards
+ * [#73131](https://github.com/rust-lang/rust/issues/73131)
+   Overall tracking Issue for `std::process` error handling
+
+## Currently-accepted wrong programs
+
+The following incorrect program fragments are all accepted today and run to completion without returning any error:
+
+```rust
+    Command::new("touch")
+        .args(&["/dev/enoent/touch-1"]);
```

Comment:
```suggestion
        .args(&["/nonexistent/touch-1"]);
```

(Rather than having to explain that `/dev/enoent` is a non-existent directory, and having people wonder if because it's under `/dev` it has some nonstandard semantics attached.)